functions.php

```diff
@@ -112,7 +115,76 @@ function get_instagram_feed($instagram_handle) {
     }
 
   }
-  //delete_transient( 'instagram_feed');
+  return $feed;
+}
+
+// Twitter Feed
+function get_twitter_feed($twitter_handle) {
+  
+  if( empty($twitter_handle) ) {
+    return new WP_ERROR('no-twitter-handle', 'Missing twitter handle');
+  }
+
+  $feed = get_transient( 'twitter_feed_' . $twitter_handle );
+
+  // If feed is not cached
+  if( empty($feed) ) {
+
+    // Require TwitterOAuth lib
+     if ( ! class_exists('TwitterOAuth')) {
+      require "lib/twitteroauth/autoload.php";
+     }
+
+    // Get keys
+    $twitter_key = IGV_get_option( '_igv_twitter_key' ); 
+    $twitter_secret = IGV_get_option( '_igv_twitter_secret' ); 
+
+    // Connect to twitter
+    $twitter = new TwitterOAuth($twitter_key, $twitter_secret);
+
+    // Get Timeline
+    $feed = $twitter->get('statuses/user_timeline', array(
+      'count'         =>  10,
+      'screen_name'   =>  $twitter_handle,
+      'trim_user'     => 'true',
+      'exclude_replies' => 'true',
+      'contributor_details' => 'false',
+      'include_rts'   => 'false'
+    ));
+
+    if( $feed->errors ) {
+      return false;
+    }
+
+    // Regex for URLs
+    $url_regex = "@(https?://([-\w\.]+[-\w])+(:\d+)?(/([\w/_\.#-]*(\?\S+)?[^\.\s])?)?)@";
```

Comment:
why not use the part of the tweet metadata that has all the links/assets in it rather than checking inside the tweet text?